### PR TITLE
fix: Updating deps to keep up with typescript

### PR DESCRIPTION
**Pull request by DavidTanner**
ride

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,22 +16,19 @@
...
+    "eslint-config-eslint": "^8.0.0",
+    "eslint-plugin-eslint-plugin": "^5.1.1",
     "eslint-plugin-node": "^11.1.0",
-    "jest": "^29.3.1",
-    "semantic-release": "^19.0.5"
-  },
-  "peerDependencies": {
-    "eslint": "^8"
+    "jest": "^29.6.1",
+    "semantic-release": "^21.0.7"
```
Comment (shawnzhu) [inline package.json:24]: > error semantic-release@21.0.7: The engine "node" is incompatible with this module. Expected version ">=18". Got "16.20.1"

That's the reason it keeps `semantic-release` to v19.

Happy to require node v18 if updating `.github/workflows/*.yaml` as well

File: lib/javascript.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ module.exports = {
     'no-unused-vars': 'error',
     'no-undef': 'error',
     'no-constant-condition': 'error',
+    'no-confusing-arrow': 'error',
```
Comment (shawnzhu) [inline lib/javascript.js:24]: This is not about dependencies.

better exclude this line into another PR to get feedback.

**Review by shawnzhu — COMMENTED** [review]
all other deps upgrades are awesome!

File: package.json (hunk 1 above)
Comment (DavidTanner) [reply package.json:24]: I updated the build to use 18.

File: lib/javascript.js (hunk 2 above)
Comment (DavidTanner) [reply lib/javascript.js:24]: Sure, I just saw it as I noticed the `prefer-arrow` plugin is quite old and thought we might be able to remove it.

**Review by shawnzhu — APPROVED** [review]

**Review by jkdowdle — APPROVED** [review]

**Pull request merged**